src/structures/interfaces/Collector.js

```diff
@@ -37,6 +38,13 @@ class Collector extends EventEmitter {
      */
     Object.defineProperty(this, 'client', { value: client });
 
+    /**
+     * The filter applied to this collector
+     * @type {CollectorFilter}
+     * @returns {boolean|Promise<boolean>}
```

Comment:
This...is not handled everywhere if it's a promise..

Reply:
It's always been like that, I just copied it over from the `CollectorFilter` typedef to make eslint happy. If it's not too far out of scope I can add the missing handling in this PR

Reply:
It's fine to put it into this PR.